modules/services/window-managers/i3.nix

```diff
@@ -378,53 +378,79 @@ let
         '';
       };
 
+      modifier = mkOption {
+        type = types.enum [ "Shift" "Control" "Mod1" "Mod2" "Mod3" "Mod4" "Mod5" ];
+        default = "Mod1";
+        description = "Modifier key that is used for all keybindings.";
+        example = "Mod4";
+      };
+
+      extraKeybindings = mkOption {
+        type = types.attrs;
+        default = {};
+        defaultText = "Additional i3 keybindings.";
+        description = ''
+          An attribute set that assignes key press to an action using key symbol.
+          See <link xlink:href="https://i3wm.org/docs/userguide.html#keybindings"/>.
+          This should be used if you want to use the default keybindings and
+          just add some more.
+        '';
+        example = literalExample ''
+          {
+            "${cfg.config.modifier}+Return" = "exec i3-sensible-terminal";
+            "${cfg.config.modifier}+Shift+q" = "kill";
+            "${cfg.config.modifier}+d" = "exec ${pkgs.dmenu}/bin/dmenu_run";
```

Comment:
won't we have the same problem of having pkgs.dmenu be built here ? like we had in a similar issue ?